File: app/src/main/java/org/deiverbum/app/data/entity/LHHymnEntity.java
package org.deiverbum.app.data.entity;

import static org.deiverbum.app.utils.Constants.LH_HYMN;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.Index;
import androidx.room.PrimaryKey;

/**
 * @author A. Cedano
 * @version 1.0
 * @since 2022.2
 */

@Entity(tableName = LH_HYMN,
        indices = {@Index(value = {"hymn"}, unique = true)}
)

public class LHHymnEntity {
    @NonNull
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "hymnID")
    public Integer himnoId = 0;

    @NonNull
    @ColumnInfo(name = "hymn")
    public String himno = "";

    @NonNull
    public String getHimno() {
        return himno;
    }
}

File: app/src/main/java/org/deiverbum/app/data/entity/LHHymnEntity.kt
package org.deiverbum.app.data.entity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Index
import androidx.room.PrimaryKey
import org.deiverbum.app.util.Constants

/**
 * Entidad para la tabla **`lh_hymn`** de la base de datos, que se ocupa de los Himnos de la Liturgia de las Horas.
 *
 * @author A. Cedano
 * @version 1.0
 * @since 2023.1
 */
@Entity(tableName = Constants.LH_HYMN, indices = [Index(value = ["hymn"], unique = true)])
data class LHHymnEntity(
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "hymnID")
    var himnoId: Int,

    @ColumnInfo(name = "hymn")
    var himno: String
)